packages/dashboards/src/lib/configurator/components/widgets/table/columns-editor-v2/table-columns-configuration-v2.component.ts

```diff
@@ -0,0 +1,281 @@
+        this.form = this.formBuilder.group({
+            columns: this.formBuilder.array([]),
+        });
+
+        this.form.controls["columns"].valueChanges
+            .pipe(takeUntil(this.onDestroy$))
+            .subscribe((columns: ITableWidgetColumnConfig[]) => {
+                this.isWidthMessageDisplayed = this.getWidthMessageDisplayed(columns);
+                this.changeDetector.markForCheck();
+            });
+
+        this.eventBus.subscribeUntil(DATA_SOURCE_CREATED, this.onDestroy$, (event: IEvent<IDataSource>) => {
+            if (!event.payload) {
+                return;
+            }
+
+            this.dataSource = event.payload;
+            this.changeDetector.markForCheck();
+        });
+
+        this.eventBus.subscribeUntil(DATA_SOURCE_OUTPUT, this.onDestroy$, (event: IEvent<any | IDataSourceOutput<any>>) => {
+            if (event.payload?.error) {
+                return;
+            }
+            const { dataFields } = isUndefined(event.payload.result) ? event.payload : (event.payload.result || {});
+            this.dataSourceFields = dataFields;
+
+            const disableColumnGeneration = this.dataSource?.features?.getFeatureConfig(WellKnownDataSourceFeatures.DisableTableColumnGeneration)?.enabled;
+
+            const columns = this.mergeColumns(this.dataSourceFields, this.getColumns());
+            if (columns?.length || disableColumnGeneration) {
+                // if disableColumnGeneration is enabled and there are no columns to be kept,
+                // we have to clear the form by updating the column input via the `onItemsChange` method
+                this.updateColumns(columns);
+            } else {
+                if (!disableColumnGeneration) {
+                    this.resetColumns(false);
+                }
+            }
+            this.changeDetector.markForCheck();
+        });
+        dataSourceManager.error.subscribe((err) => {
+            this.dataSourceError = err;
+            this.changeDetector.markForCheck();
+        });
+    }
+
+    public ngOnInit() {
+        this.formReady.emit(this.form);
+
+        if (this.template && !this.tableFormatterRegistry.getItems()?.length) {
+            const formatters = this.template?.[1].properties?.formatters;
+            this.tableFormatterRegistry.addItems(formatters);
+        }
+    }
+
+    public ngOnChanges(changes: SimpleChanges): void {
+        // columns from input property are set directly to the form
+        if (changes.columns) {
+            this.updateColumns(this.columns, false);
+        }
+    }
+
+    public ngOnDestroy(): void {
+        this.onDestroy$.next();
+        this.onDestroy$.complete();
+    }
+
+    public getColumns() {
+        return this.form.controls["columns"].value;
+    }
+
+    public updateColumns(columns: ITableWidgetColumnConfig[], emitEvent: boolean = true) {
+        const arr = this.form.controls["columns"] as FormArray;
```

Comment:
would be slightly more readable as:
```suggestion
        const columns = this.form.controls["columns"] as FormArray;
```

Reply:
Input is named columns already

Reply:
I don't think that will conflict since the input would be referred to as `this.columns` and this would just be `columns`.

Reply:
Sorry its an argument of the function see line 149